indy_node/server/client_authn.py

```diff
@@ -1,35 +1,41 @@
 from copy import deepcopy
 from hashlib import sha256
 
-from plenum.common.exceptions import UnknownIdentifier
 from plenum.common.types import OPERATION
-from plenum.common.constants import TXN_TYPE, RAW, ENC, HASH, VERKEY
-from plenum.server.client_authn import NaclAuthNr
+from plenum.common.constants import TXN_TYPE, RAW, ENC, HASH
+from plenum.server.client_authn import NaclAuthNr, CoreAuthNr, CoreAuthMixin
 
-from indy_common.constants import ATTRIB
+from indy_common.constants import ATTRIB, POOL_UPGRADE, SCHEMA, CLAIM_DEF, \
+    GET_NYM, GET_ATTR, GET_SCHEMA, GET_CLAIM_DEF, POOL_CONFIG
 from indy_node.persistence.idr_cache import IdrCache
-# from indy_node.persistence.state_tree_store import StateTreeStore
 
 
-class TxnBasedAuthNr(NaclAuthNr):
+class LedgerBasedAuthNr(CoreAuthMixin, NaclAuthNr):
     """
     Transaction-based client authenticator.
     """
 
+    write_types = CoreAuthMixin.write_types.union({ATTRIB, SCHEMA, CLAIM_DEF,
+                                                   POOL_CONFIG, POOL_UPGRADE})
+    query_types = CoreAuthMixin.query_types.union({GET_NYM, GET_ATTR, GET_SCHEMA,
```

Comment:
Can we get `query_types` from registered req handlers? Looks like we have a duplication here (query_types in both Requesthandlers and LedgerBasedAuthNr). Moreover, if new req handlers are added, then we need to change `query_types` in LedgerBasedAuthNr manually.

Reply:
Ok